httpx/models.py

```diff
@@ -100,7 +100,7 @@ def __init__(
             if idna_authority != self._uri_reference.authority:
                 self._uri_reference = self._uri_reference.copy_with(
                     authority=idna_authority
-                )
+                )  # pragma: nocover
```

Comment:
Instead of disabling coverage here let's create a simple unit test for the branch.

Reply:
I couldn't find a test case because if I try to Unicode domain, I get this error: `rfc3986.exceptions.InvalidAuthority: The authority (😁.de) is not valid.` 

Should we change this part if there is no case or could you help me with this?

Reply:
What about `ουτοπία.δπθ.gr`? Emojis aren't valid for IDNs for IDNA 2008.

Reply:
idna encode raises an error if there is any char that greater then 128. (it's not just about emoji [link](https://github.com/python-hyper/rfc3986/blob/2982509402519575282845adf1f39dc0195369e0/src/rfc3986/iri.py#L121)) so this if clause is never used. If you want we can change it with another PR and check `InvalidAuthority` error?

Reply:
I think we can literally just get rid of that whole `if` statement branch. `IRIReference.encode()` does everything for us.

Reply:
added a pr #210